partner_firstname/tests/test_onchange.py

```diff
@@ -6,99 +6,95 @@
 The form operates in onchange mode, with its limitations.
 """
 
+from odoo.tests.common import Form
+
 from .base import OnChangeCase
 
 
 class PartnerCompanyCase(OnChangeCase):
     is_company = True
 
     def test_create_from_form(self):
-        """A user creates a company from the form."""
         name = "Sôme company"
-        with self.env.do_in_onchange():
-            # User presses ``new``
-            partner = self.new_partner()
-
-            # User sets a name, which triggers onchanges
-            partner.name = name
-            partner._onchange_name()
```

Comment:
@rven I don't see why you should use a Form to call the onchange. If you get an error when calling the onchange directly, it's maybe a bug into the orm... We should at least understand the root cause of the issue into the original code. I already encountered some errors when calling an onchange method with computed fields when the compute method makes an SQL query with the `id` of the record as parameter without checking that the id is not a NewId....

Reply:
Hi @lmignon,

I don't think you're reviewing my changes.
Your suggestion is also labeled **Outdated**
I don't have any calls to _onchange_name, that's why I changed it to a Form instance.